client/homebrew/pages/editPage/editPage.jsx

```diff
@@ -52,35 +57,81 @@ const EditPage = createClass({
 
 	getInitialState : function() {
 		return {
-			brew                   : this.props.brew,
+			brew : {
+				text        : this.props.brew.text || '',
+				style       : this.props.brew.style || '',
+				shareId     : this.props.brew.shareId || null,
+				editId      : this.props.brew.editId || null,
+				gDrive      : this.props.brew.gDrive || false,
+				title       : this.props.brew.title || '',
+				description : this.props.brew.description || '',
+				tags        : this.props.brew.tags || '',
+				published   : this.props.brew.published || false,
+				authors     : this.props.brew.authors || [],
+				systems     : this.props.brew.systems || [],
+				renderer    : this.props.brew.renderer || 'legacy'
+			},
 			isSaving               : false,
 			isPending              : false,
 			alertTrashedGoogleBrew : this.props.brew.trashed,
 			alertLoginToTransfer   : false,
-			saveGoogle             : this.props.brew.googleId ? true : false,
+			saveGoogle             : (global.account && global.account.googleId ? true : false),
 			confirmGoogleTransfer  : false,
 			errors                 : null,
 			htmlErrors             : Markdown.validate(this.props.brew.text),
-			url                    : ''
+			url                    : '',
+			editType               : ''
 		};
 	},
 	savedBrew : null,
 
 	componentDidMount : function(){
-		this.setState({
-			url : window.location.href
+		// Determine editor type
+		editTypes.forEach((type)=>{
+			if(this.props.editType == type) {
+				this.editType = this.props.editType;
+			};
 		});
+		if(!this.editType){
+			// ERROR: Unknown editType
+			console.log('Unknown EditType parameter!');
+			// Redirect to Home page; need to revisit with better Error messaging in the future
+			window.location.href = `/`;
+		};
```

Comment:
See above comment. All of these checks aren't necessary. We also don't need to worry about redirecting at this stage. That is already handled in the React router (if a user visits anything that is an invalid path, it takes them to the home page).
```suggestion
```